daemon/main.go

```diff
@@ -615,7 +615,12 @@ func initEnv(cmd *cobra.Command) {
 	policy.SetPolicyEnabled(strings.ToLower(viper.GetString("enable-policy")))
 
 	if err := kvstore.Setup(kvStore, kvStoreOpts); err != nil {
-		log.WithError(err).Fatal("Unable to setup kvstore")
+		addrkey := fmt.Sprintf("%s.address", kvStore)
+		addr := kvStoreOpts[addrkey]
```

Comment:
Not sure whether this is going to properly retrieve the address that is being used to contact the kvstore.

Reply:
Looking at https://github.com/cilium/cilium/blob/d58bf6c3703a8f053bb54cce13245a8a4d1e91f8/pkg/option/config.go#L78-L83 and how we create that particular option (without a validator), I expect that this will work if the CLI option looked like `--kvstore-opts etcd.address=something`. I'm not sure if we validate that elsewhere to make sure it conforms though :/

Reply:
I think we should add a validator there.